lzip.go

```diff
@@ -0,0 +1,52 @@
+package archiver
+
+import (
+	"bytes"
+	"io"
+	"strings"
+
+	"github.com/sorairolake/lzip-go"
+)
+
+func init() {
+	RegisterFormat(Lzip{})
+}
+
+// Lzip facilitates lzip compression.
+type Lzip struct{}
+
+func (Lzip) Name() string { return ".lz" }
+
+func (lz Lzip) Match(filename string, stream io.Reader) (MatchResult, error) {
+	var mr MatchResult
+
+	// match filename
+	if strings.Contains(strings.ToLower(filename), lz.Name()) {
```

Comment:
Ah, indeed -- make this comparison more strict and that should hopefully fix the test.

Reply:
I think this can be fixed with the following changes:

```suggestion
	if filepath.Ext(strings.ToLower(filename)) == lz.Name() {
```

However, this requires that the extension be the last element of the path.

Reply:
That would fail for `.lzip.gz` as you say, but maybe a check for both a suffix of `.lz` or the string contains `.lz.`?

Reply:
I think there is almost no possibility that `.lz` and `.lz4` exists anywhere other than at the end of the path.

Reply:
That's a good point.